### PR TITLE
feat: add wf-controller healthz param

### PR DESCRIPTION
**Pull request by yyzxw**
Fixes #TODO

### Motivation

In some cases, the controller may restart frequently, thus modifying the survival probe

### CODE REVIEW

File: workflow/controller/healthz.go [hunk 1]
```diff
@@ -20,6 +20,14 @@ var (
 // https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request
 // If we are in a state where there are any workflows that have not been reconciled in the last 2m, we've gone wrong.
 func (wfc *WorkflowController) Healthz(w http.ResponseWriter, r *http.Request) {
+	// If the param "full" is set to "false", only check if the controller is running.
```
Comment (terrytangyuan) [inline workflow/controller/healthz.go:23]: Is there any reference?

Comment (yyzxw) [reply workflow/controller/healthz.go:23]: yes, reference to argo-cd(https://github.com/argoproj/argo-cd/blob/master/cmd/argocd-repo-server/commands/argocd_repo_server.go#L136),

**Comment by juliev0** [comment]
So sorry, I'm going through old PRs and I see that unfortunately this one has been forgotten. I will try to review it soon.

Argo Workflows needs more people approving PRs unfortunately. If your company is by chance interested, I am gauging interest in [this](https://github.com/argoproj/argo-workflows/issues/12229) idea.

**Review by Joibel — CHANGES_REQUESTED** [review]
I know the healthz endpoint isn't documented anywhere at the moment, but this feels like it would be a hidden feature with an even more hidden application.
Please could you document this feature and when it might be useful somewhere appropriate in /docs.

**Comment by juliev0** [comment]
Closing this PR as it's been stale for awhile

**Pull request closed without merging**